pkg/kubelet/status/state/checkpoint.go

```diff
@@ -18,48 +18,62 @@ package state
 
 import (
 	"encoding/json"
-
-	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/core/v1"
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/checksum"
 )
 
-var _ checkpointmanager.Checkpoint = &PodResourceAllocationCheckpoint{}
+var _ checkpointmanager.Checkpoint = &Checkpoint{}
 
-// PodResourceAllocationCheckpoint is used to store resources allocated to a pod in checkpoint
-type PodResourceAllocationCheckpoint struct {
+type PodResourceAllocationInfo struct {
 	AllocationEntries   map[string]map[string]v1.ResourceRequirements `json:"allocationEntries,omitempty"`
 	ResizeStatusEntries map[string]v1.PodResizeStatus                 `json:"resizeStatusEntries,omitempty"`
-	Checksum            checksum.Checksum                             `json:"checksum"`
 }
 
-// NewPodResourceAllocationCheckpoint returns an instance of Checkpoint
-func NewPodResourceAllocationCheckpoint() *PodResourceAllocationCheckpoint {
-	//lint:ignore unexported-type-in-api user-facing error message
-	return &PodResourceAllocationCheckpoint{
-		AllocationEntries:   make(map[string]map[string]v1.ResourceRequirements),
-		ResizeStatusEntries: make(map[string]v1.PodResizeStatus),
+// Checkpoint represents a structure to store pod resource allocation checkpoint data
+type Checkpoint struct {
+	// Data is a JSON serialized checkpoint data
+	Data string `json:"data"`
```

Comment:
No particular reason, just following existing code in dra checkpoint [link](https://github.com/kubernetes/kubernetes/blob/ce47f7b416ff0e5a5c73d70e8dd3175ced5afaa9/pkg/kubelet/cm/dra/state/checkpoint.go#L34). What do you think? Which is better?

Reply:
`[]byte` seems more idiomatic, and makes the intent clearer. No strong preference though.

That said, a lot of this checkpoint code is identical to the DRA checkpoint, including this `Checkpoint` type. Should we just move all of this to the pkg/kubelet/checkpointmanager package?

Reply:
> That said, a lot of this checkpoint code is identical to the DRA checkpoint, including this Checkpoint type. Should we just move all of this to the pkg/kubelet/checkpointmanager package?

Could we separate this into a different task? Since there are many checkpoints, cpu, memory, dra, device and status, refactoring the code will impact not only InPlacePodVerticalScaling feature gate.

Reply:
What do you think? @tallclair

Reply:
Yeah, sounds good. Can you file an issue and/or leave a TODO?